java/configuration/src/test/java/sleeper/configuration/properties/InstancePropertiesTest.java

```diff
@@ -284,4 +286,37 @@ private static InstanceProperties getSleeperProperties() {
 
         return instanceProperties;
     }
+
+    @Test
+    void shouldGetAllUserDefinedAndSystemDefinedProperties() {
+        // Given/When
+        List<InstanceProperty> propertyList = InstanceProperty.getAllProperties();
+
+        // Then
+        assertThat(propertyList)
+                .containsAll(Arrays.asList(UserDefinedInstanceProperty.values()));
+        assertThat(propertyList)
+                .containsAll(Arrays.asList(SystemDefinedInstanceProperty.values()));
+    }
+
+    @Test
+    void shouldOrderAllPropertiesBasedOnGroup() {
+        // Given/When
+        List<InstanceProperty> propertyList = InstanceProperty.getAllGroupedProperties();
+
+        // Then
+        // Order COMMON before INGEST
+        assertThat(propertyList.indexOf(UserDefinedInstanceProperty.ID))
+                .isLessThan(propertyList.indexOf(UserDefinedInstanceProperty.ECR_INGEST_REPO));
+        // Order BULK_IMPORT before PARTITION_SPLITTING
+        assertThat(propertyList.indexOf(UserDefinedInstanceProperty.BULK_IMPORT_CLASS_NAME))
+                .isLessThan(propertyList.indexOf(UserDefinedInstanceProperty.PARTITION_SPLITTING_PERIOD_IN_MINUTES));
+
+        // Order COMMON before INGEST
+        assertThat(propertyList.indexOf(SystemDefinedInstanceProperty.CONFIG_BUCKET))
+                .isLessThan(propertyList.indexOf(SystemDefinedInstanceProperty.INGEST_CLUSTER));
+        // Order BULK_IMPORT before PARTITION_SPLITTING
+        assertThat(propertyList.indexOf(SystemDefinedInstanceProperty.BULK_IMPORT_BUCKET))
+                .isLessThan(propertyList.indexOf(SystemDefinedInstanceProperty.PARTITION_SPLITTING_QUEUE_URL));
```

Comment:
It would be good if we can somehow assert that all of the system and user defined properties are present in the exact correct order, as well as that all properties of each group are together and that the groups are ordered correctly.